code/modules/mob/living/carbon/monkey/diona.dm

```diff
@@ -135,6 +136,17 @@
 				return 1
 	return ..()
 
+/mob/proc/sample_can_reach(mob/M as mob, sting_range = 1)
+	if(M.loc == src.loc)
+		return 1 //target and source are in the same thing
+	if(!isturf(src.loc) || !isturf(M.loc))
+		return 0 //One is inside, the other is outside something.
+	if(sting_range < 2)
```

Comment:
This can be trimmed down to just "return Adjacent(M)" if you're not doing variable sampling ranges.